kornia/contrib/prompter.py

```diff
@@ -0,0 +1,374 @@
+
+        # Embed prompts
+        sparse_embeddings, dense_embeddings = self.model.prompt_encoder(
+            points=prompts.points, boxes=prompts.boxes, masks=prompts.masks
+        )
+        del prompts
+
+        # Predict masks
+        logits, scores = self.model.mask_decoder(
+            image_embeddings=self.image_embeddings,
+            image_pe=self.model.prompt_encoder.get_dense_pe(),
+            sparse_prompt_embeddings=sparse_embeddings,
+            dense_prompt_embeddings=dense_embeddings,
+            multimask_output=multimask_output,
+        )
+        results = SegmentationResults(logits, scores)
+        if (
+            output_original_size
+            and isinstance(self._input_image_size, tuple)
+            and isinstance(self._original_image_size, tuple)
+        ):
+            results.original_res_logits(self._input_image_size, self._original_image_size, self._input_encoder_size)
+
+        results = results.squeeze(0)
+
+        return results
+
+    def reset_image(self) -> None:
+        self._tfs_params = None
+        self._original_image_size = None
+        self._input_image_size = None
+        self._input_encoder_size = None
+
+        if hasattr(self, 'image_embeddings'):
+            del self.image_embeddings
+
+        self.image_embeddings = None
+        self.is_image_set = False
+
+    def compile(
+        self,
+        *,
+        fullgraph: bool = False,
+        dynamic: bool = False,
+        backend: str = 'inductor',
+        mode: str | None = None,
+        options: dict[Any, Any] = {},
+        disable: bool = False,
+    ) -> None:
+        """Applies `torch.compile(...)`/dynamo API into the ImagePrompter API.
+
+        .. note:: For more information about the dynamo API check the official docs
+                  https://pytorch.org/docs/stable/generated/torch.compile.html
+
+        Args:
+            fullgraph: Whether it is ok to break model into several subgraphs
+            dynamic: Use dynamic shape tracing
+            backend: backend to be used
+            mode: Can be either “default”, “reduce-overhead” or “max-autotune”
+            options: A dictionary of options to pass to the backend.
+            disable: Turn torch.compile() into a no-op for testing
+
+        Example:
+            >>> # prompter = ImagePrompter()
+            >>> # prompter.compile() # You should have torch >= 2.0.0 installed
+            >>> # Use the prompter methods ...
+        """
+
+        # self.set_image = torch.compile(  # type: ignore[method-assign]
+        #     self.set_image,
+        #     fullgraph=fullgraph,
+        #     dynamic=dynamic,
+        #     backend=backend,
+        #     mode=mode,
+        #     options=options,
+        #     disable=disable,
+        # )
+        # FIXME: compile set image will try to compile AugmentationSequential which fails
+        self.model.image_encoder = torch.compile(  # type: ignore
+            self.model.image_encoder,
```

Comment:
```suggestion
            self.model,
```
why not directly this ?

Reply:
torch `compile` returns a callable, so, if we compile a `Module` we can just use the `forward`, and can't access its methods or properties -- at the prompter we want to use the encoder and decoder separately from the model